### PR TITLE
[gradle] Add DSL to configure compose resources

### PR DESCRIPTION
**Pull request by terrakok**
Example:
```kotlin
compose.resources {
    publicResClass = true
    packageOfResClass = "me.sample.library.resources"
    generateResClass = auto
}
```

### CODE REVIEW

**Comment by igordmn** [comment]
Adding a second reviewer, as we introduce a new API

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ResourcesExtension.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    var publicResClass: Boolean = false
+
+    /**
+     * The unique identifier of the resources in the current project.
+     * Uses as package for the generated Res class and for isolation resources in a final artefact.
+     *
+     * If it is empty then `{group name}.{module name}.generated.resources` will be used.
+     *
+     */
+    var resourceProjectId: String = ""
```
[the remark this answers is not in the document]
Comment (terrakok) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ResourcesExtension.kt:18]: For DSLs it is not convenient to have a gradle provider. And it is not required here, cause the valve is not computable by the user's perspective.

If we have a null as default we need to check the empty case as well

[the remark this answers is not in the document]
Comment (igordmn) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/ResourcesExtension.kt:18]: Okay, it is not very important.

**Review by igordmn — APPROVED** [review]

**Review by pjBooms — APPROVED** [review]

**Comment by terrakok** [comment]
Renamed DSL parameter to make it more clear

```diff
compose.resources {
    publicResClass = true
-     resourceProjectId = "me.sample.library.resources"
+     packageOfResClass = "me.sample.library.resources"
    generateResClass = auto
}
```

**Comment by Deorigami** [comment]
Hi @terrakok ,, sorry im still confused, how to set the Res class become public? i already use 1.6.2, but my Res class still internal, and i cant find the compose.resource extentions in my gradle, thanks if you reply this

**Comment by mdsadique-inam** [comment]
Hi @terrakok :wave: I can make the Res class public and access my string Res from another project in code, but if I don't keep the copy of the same string resource, then the UI will show only an empty string.

**Comment by m0rtis** [comment]
Many-many thanx!!!

**Comment by terrakok** [comment]
The feature works only with kotlin 2.0.0

**Comment by mdsadique-inam** [comment]
Hello @terrakok :wave: with kotlin version 2.0.0-RC1 development is working well but still same issue in production

**Comment by mdsadique-inam** [comment]
@terrakok this is my project repo https://github.com/mdsadique-inam/PDNS-Manager you can build using `./gradlew :launchpad:wasmJsBrowserProductionWebpack` you may notice the values folder inside the built output-dir is empty

**Comment by terrakok** [comment]
OK. this is not related with the DSL. It seems a kotlin wasm target's problem.

**Comment by terrakok** [comment]
https://youtrack.jetbrains.com/issue/KT-66228
the corresponding issue. it is marked as fixed. I notified the team to recheck it again

**Comment by eymar** [comment]
> @terrakok this is my project repo https://github.com/mdsadique-inam/PDNS-Manager you can build using `./gradlew :launchpad:wasmJsBrowserProductionWebpack` you may notice the values folder inside the built output-dir is empty and the resources are coming from `commonCompose`

please have a look at: https://youtrack.jetbrains.com/issue/KT-66228/K-Wasm-2.0.0-Beta4-distribution-doesnt-contain-all-files#focus=Comments-27-9719548.0-0

**Comment by mdsadique-inam** [comment]
@terrakok @eymar thank you for providing your precious to me.

**Comment by mdsadique-inam** [comment]
wait what? I don't want to open this PR I only want that person to resolve their issue

**Pull request merged**